topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -418,12 +423,17 @@ private void updateInternal(List<ReconcilableTopic> topics) {
 
         Map<ReconcilableTopic, Either<TopicOperatorException, Object>> results = new HashMap<>();
 
-        var partitionedByManaged = partitionedByDeletion.get(false).stream().collect(Collectors.partitioningBy(reconcilableTopic -> isManaged(reconcilableTopic.kt())));
+        var partitionedByManaged = partitionedByDeletion.get(false).stream()
+            .collect(Collectors.partitioningBy(reconcilableTopic -> isManaged(reconcilableTopic.kt())));
         var unmanaged = partitionedByManaged.get(false);
         addOrRemoveFinalizer(useFinalizer, unmanaged).forEach(rt -> putResult(results, rt, Either.ofRight(null)));
 
-        List<ReconcilableTopic> mayNeedUpdate = validateManagedTopics(partitionedByManaged);
+        // skip reconciliation of paused KafkaTopics
+        var partitionedByPaused = validateManagedTopics(partitionedByManaged).stream()
+            .collect(Collectors.partitioningBy(reconcilableTopic -> isPaused(reconcilableTopic.kt())));
```

Comment:
We've now got three sequential `partitioningBy()`s going on. I wonder if the code would be simpler if we just did a `groupingBy()` instead?